libc/string_lenchr.c

```diff
@@ -96,61 +99,255 @@ TEST(string_len, ascii)
+	memset(bigStr, 1, sizeof(bigStr) - 1);
+
+	sz = sizeof(bigStr) - 1;
+	TEST_ASSERT_EQUAL_INT(sz, strlen(bigStr));
+	TEST_ASSERT_EQUAL_INT(sz - 1, strnlen(bigStr, sz - 1));
+	TEST_ASSERT_EQUAL_INT(sz, strnlen(bigStr, sz));
+	TEST_ASSERT_EQUAL_INT(sz, strnlen(bigStr, sz + 1));
+}
+
+/*
+////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_spn)
+{
+}
+
+
+TEST_TEAR_DOWN(string_spn)
+{
+}
+
+
+TEST(string_spn, ascii)
+{
+	char supportCharSet[BUFF_SIZE] = { 0 },
+		 *asciiStr;
+	int i;
+
+	asciiStr = testdata_createCharStr(BUFF_SIZE + 1);
+
+
+	for (i = 1; i < BUFF_SIZE; i++) {
+		supportCharSet[i - 1] = i;
+		/*
+		 * In this case we need to use fully filled set for strcspn
+		 * because it counts size based on elements that are not
+		 * present in himself
+		 */
+		TEST_ASSERT_EQUAL_INT(i - 1, strcspn(&asciiStr[1], &asciiStr[i]));
+		TEST_ASSERT_EQUAL_INT(i, strspn(&asciiStr[1], supportCharSet));
+	}
+
+	free((void *)asciiStr);
+}
+
+
+TEST(string_spn, not_ascii)
+{
+	unsigned char charSet[BUFF_SIZE] = { 0 };
 	int sz, i;
 
-	/* The length of string is not clearly restricted, so we test one of bigger value, which may be used  */
-	for (i = 0; i < PATH_MAX - 1; i++) {
-		bigstr[i] = 'A';
+	/* Checking  out of ASCII bytes */
+	for (i = sizeof(charSet); i < sizeof(charSet) * 2 - 1; i++) {
+		charSet[i - sizeof(charSet)] = i;
 	}
-	bigstr[i] = '\0';
+
+	sz = sizeof(charSet) - 1;
+	TEST_ASSERT_EQUAL_INT(sz, strcspn((const char *)charSet, ""));
+	TEST_ASSERT_EQUAL_INT(sz, strspn((const char *)charSet, (const char *)charSet));
+}
+
+
+TEST(string_spn, big)
+{
+	char bigstr[PATH_MAX] = { 0 };
+	int sz;
+
+	/*
+	 * The length of the string is not restricted,
+	 * so we test one of the bigger values, which may be used
+	 * with remembering the last element must be a null term
+	 * zero to collect the data as the string from
+	 * declared space, that's why we memset one place less than its size
+	 */
+	memset(bigstr, 'a', sizeof(bigstr) - 1);
 
 	sz = sizeof(bigstr) - 1;
-	TEST_ASSERT_EQUAL_INT(sz, strlen(bigstr));
-	TEST_ASSERT_EQUAL_INT(sz - 1, strnlen(bigstr, sz - 1));
-	TEST_ASSERT_EQUAL_INT(sz, strnlen(bigstr, sz));
-	TEST_ASSERT_EQUAL_INT(sz, strnlen(bigstr, sz + 1));
+	TEST_ASSERT_EQUAL_INT(sz, strcspn(bigstr, ""));
+	TEST_ASSERT_EQUAL_INT(sz, strspn(bigstr, bigstr));
+}
+
+
+TEST(string_spn, empty_output)
+{
+	char emptySet[BUFF_SIZE] = { 0 };
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn(emptySet, emptySet));
+	TEST_ASSERT_EQUAL_INT(0, strspn(emptySet, emptySet));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn(emptySet, ""));
+	TEST_ASSERT_EQUAL_INT(0, strspn(emptySet, ""));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn(emptySet, "abc"));
+	TEST_ASSERT_EQUAL_INT(0, strspn(emptySet, "abc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("", "abc"));
+	TEST_ASSERT_EQUAL_INT(0, strspn("", "abc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("", ""));
+	TEST_ASSERT_EQUAL_INT(0, strspn("", ""));
```

Comment:
As `s1` is a string it does not matter, whether there is one or more NUL terminators.

```suggestion
	TEST_ASSERT_EQUAL_INT(0, strcspn("", "abc"));
	TEST_ASSERT_EQUAL_INT(0, strspn("", "abc"));

	TEST_ASSERT_EQUAL_INT(0, strcspn("", ""));
	TEST_ASSERT_EQUAL_INT(0, strspn("", ""));
```